includes/class-wcs-notifications-batch-processor.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+use Automattic\WooCommerce\Internal\BatchProcessing\BatchProcessingController;
+use Automattic\WooCommerce\Internal\BatchProcessing\BatchProcessorInterface;
```

Comment:
Just noting that these are `Internal` classes from WC Core. In general, I don't think we should be using `Internal` classes outside of WC Core, since they could be changed in non-backward compatible ways without consideration that there might be other extensions using them (hence why they are `Internal`).

Reply:
Thanks for raising this, Matt, I initially missed that. Perhaps even more importantly, this won't work in WC before 9.0 as the way core dependency injection container loaded the BatchProcessor wouldn't allow it to use our class (which is outside of WC core's Internal namespace). This problem was fixed by [this PR](https://github.com/woocommerce/woocommerce/pull/46975), but since WC Subscriptions need to support WC 7.7+, it's a no-go for us. @xristos3490 is working on pulling relevant bits of batch processing from core to subscriptions core instead.

Reply:
Yes, thanks, Matt! This might apply to your thinking around gift cards. Here's the PR: 
- https://github.com/Automattic/woocommerce-subscriptions-core/pull/689